lab_2_keywords_cooccurrence/main.py

```diff
@@ -40,7 +71,12 @@ def calculate_frequencies_for_content_words(candidate_keyword_phrases: KeyPhrase
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not correct_type(candidate_keyword_phrases, list):
+        return None
+    freq_dict = {}
+    for phrase in candidate_keyword_phrases:
+        freq_dict |= {token: (freq_dict[token] + 1 if token in freq_dict else phrase.count(token)) for token in phrase}
```

Comment:
looks very interesting, but it takes time to understand this line. I think just adding to `freq_dict` is more clear